### PR TITLE
Sphinx configuration files and readme fixes

### PR DESCRIPTION
**Pull request by SylvainCorlay**
Adding the sphinx configuration files and minor update to documentation. 

I am getting a number of `Unexpected section title` warnings because we are not using the standard section names.

### CODE REVIEW

**Comment by SylvainCorlay** [comment]
Ready for review.

**Comment by SylvainCorlay** [comment]
`fill, drag_color, names_unique, enable_move, enable_add, enable_delete, restrict_x, restrict_y, update_on_move`
are undocumented attributes of Scatter.

**Comment by SylvainCorlay** [comment]
@jasongrout @ssunkara1 I squashed pretty much everything in a single commit but there is a lot here. This requires sphinx 1.3 (the conda-packaged version is 1.2.3).

File: bqplot/__init__.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 .. currentmodule:: bqplot
 
 
-Each plot starts with a `Figure` object.  A `Figure` has a number of `Axis` objects (horizontal and vertical) and a number of `Mark` objects. Each `Mark` object is a visual representation of the data. Each `Axis` and `Mark` has a `Scale` object.  The `Scale` objects transform data into a visual property (typically a location in pixel space, but could be a color, etc.).  An `Axis` draws an axis associated with the scale. ::
+Each plot starts with a `Figure` object.  A `Figure` has a number of `Axis` objects (representing scales) and a number of `Mark` objects. `Mark` objects are a visual representation of the data. Scales transform data into visual properties (typically a numeber of pixels, a color, etc.).  ::
```
Comment (jasongrout) [inline bqplot/__init__.py]: numeber -> number

File: bqplot/axes.py [hunk 2]
```diff
@@ -59,46 +61,46 @@ class Axis(BaseAxis):
...
+        The side of the axis, either bottom, top, left or right.
+    label: string (default: '')
+        The axis label
+    tick_format: string or None (default: '')
+        The tick format for the axis.
+    scale: Scale
+        The scale represented by the axis
+    num_ticks: int or None (default: None)
+        If tick_values is None, number of ticks
+    tick_values: numpy.ndarray or None (default: [])
```
Comment (jasongrout) [inline bqplot/axes.py]: tick_values doesn't have `allow_none=True`.

File: bqplot/interacts.py [hunk 3]
```diff
@@ -133,15 +138,16 @@ class PanZoom(Interaction):
...
-            the corresponding direction which should be panned or zoomed.
+    Attributes
+    ----------
+
+    allow_pan: bool (default: True)
+        attribute to set the ability to pan the figure or not
+    allow_zoom: bool (default: True)
+        attribute to set the ability to zoom the figure or not
+    scales: Dictionary of lists of Scales (default: {})
+        Dictionary with keys as 'x' and 'y' and values being the scales in
```
Comment (jasongrout) [inline bqplot/interacts.py]: Dictionary with keys 'x' and 'y' and values... (delete "as")

File: bqplot/marks.py [hunk 4]
```diff
@@ -423,11 +427,10 @@ class Scatter(Mark):
         Labels for the points of the chart
     display_names: bool (default: True)
         Controls whether names are displayed for points in the scatter
-    fill, drag_color, names_unique, enable_move, enable_add,
-    enable_delete, restrict_x, restrict_y, update_on_move.
 
     Data Attributes
```
Comment (jasongrout) [inline bqplot/marks.py]: `**Data Attributes**` ?  There are also a lot of other non-supported headings below.

File: bqplot/axes.py (hunk 2 above)
Comment (SylvainCorlay) [inline bqplot/axes.py]: I think that this should be corrected in a different PR. Can you open an issue with this?

**Comment by SylvainCorlay** [comment]
We are still missing docstrings for 

```
on_background_click(callback[, remove])	
on_click(callback[, remove])	
on_element_click(callback[, remove])	
on_hover(callback[, remove])	
on_legend_click(callback[, remove])	
on_legend_hover(callback[, remove])
```

File: bqplot/axes.py (hunk 2 above)
Comment (jasongrout) [inline bqplot/axes.py]: Done: https://github.com/bloomberg/bqplot/issues/38

File: bqplot/marks.py (hunk 4 above)
Comment (SylvainCorlay) [inline bqplot/marks.py]: Napoleon disables that feature.

**Comment by jasongrout** [comment]
```
bqplot/bqplot/pyplot/pyplot.py:docstring of bqplot.pyplot.pyplot.scales:22: WARNING: Definition list ends without a blank line; unexpected unindent.
```

```
bqplot/bqplot/pyplot/pyplot.py:docstring of bqplot.pyplot.pyplot.scales:None: WARNING: more than one target found for cross-reference 'scale': bqplot.axes.Axis.scale, bqplot.axes.ColorAxis.scale, bqplot.interacts.OneDSelector.scale
```

**Comment by jasongrout** [comment]
I updated and those errors are gone.

Here are two more things from `grep -n -B 1  -- "----" *.py`:

Everywhere else, it is "Note"
```
marks.py-452-    Notes 
marks.py:453:    -----
```

Unsupported headers:
```
--
market_map.py-83-    Map Drawing Attributes
market_map.py:84:    ----------------------
--
market_map.py-111-    Display Attributes
market_map.py:112:    ------------------
--
market_map.py-126-    Other Attributes
market_map.py:127:    ----------------
```

**Comment by jasongrout** [comment]
You apparently already solved these things too.

**Comment by jasongrout** [comment]
@SylvainCorlay - are you doing any more changes?  I'm happy.
